core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java

```diff
@@ -1420,6 +1425,16 @@
 
   public static final String CARBON_UPDATE_SYNC_FOLDER_DEFAULT = "/tmp/carbondata";
 
+  /**
+   * S3 Constants
+   */
+
+  public static final String S3_IMPLEMENTATION = "fs.s3a.impl";
```

Comment:
How about S3N support? I think this can be a property set by user, user can choose S3 implementation class. The default class can be S3A